### PR TITLE
feat(invariants): add module invariants

### PR DESCRIPTION
**Pull request by Pantani**
### What does this PR does?

Add `profile`, `launch` and `campaign` module [invariants](https://docs.cosmos.network/master/building-modules/invariants.html)

### CODE REVIEW

**Review by lubtd — APPROVED** [review]
Thanks for having investigated into this👍

I think those current ones are good, though I'm wondering if the unknown request is critical to have but we can finalize all of them at the end of the Milestone

**Comment by lubtd** [comment]
For Unknow Request, I'm wondering this because too much invariants would theoretically slow down the simulation process

**Comment by Pantani** [comment]
> For Unknow Request, I'm wondering this because too much invariants would theoretically slow down the simulation process

I thought the same, but I don't know the actual cost of have invariants. We can delete the entire method or keep only the interface validate call:

```go

// UnknownRequestTypeInvariant invariant that checks if the Request
// type is valid
func UnknownRequestTypeInvariant(k Keeper) sdk.Invariant {
	return func(ctx sdk.Context) (string, bool) {
		all := k.GetAllRequest(ctx)
		for _, request := range all {
			if err := request.Content.Validate(); err != nil {
				return sdk.FormatInvariant(
					types.ModuleName, "unknown-request-type",
					"invalid request",
				), true
			}
		}
		return "", false
	}
}
```

**Comment by lubtd** [comment]
> I thought the same, but I don't know the actual cost of have invariants. We can delete the entire method or keep only the interface validate call

We can check this afterward since it's trivial to disable a invariant

File: x/launch/keeper/invariants.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/tendermint/spn/x/launch/types"
+)
+
+// RegisterInvariants registers all module invariants
+func RegisterInvariants(ir sdk.InvariantRegistry, k Keeper) {
+	ir.RegisterRoute(types.ModuleName, "zero-launch-timestamp",
```
Comment (ilgooz) [inline x/launch/keeper/invariants.go:12]: Maybe we can move routes to constants and reuse in the other part of the code?

Comment (Pantani) [reply x/launch/keeper/invariants.go:12]: I like the solution, but the methods are hard for we can reuse. There is too much specific logic for being reusable. I used the same pattern from the [cosmos-sdk modules](https://github.com/cosmos/cosmos-sdk/blob/master/x/bank/keeper/invariants.go).

**Comment by Pantani** [comment]
added invariants for the campaign module too

**Review by lubtd — APPROVED** [review]

**Review by ilgooz — APPROVED** [review]

**Pull request merged**